### PR TITLE
Select no value change on focusout

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 🤨 Rationale

Consensus was reached that it felt incorrect for the `Select` to update its value to something different after navigating to a new option and then either pressing `<Tab>` or clicking away.

## 👩‍💻 Implementation

Changed it so that the `focusoutHandler` no longer updates the `selectedIndex`.

## 🧪 Testing

Changed the tests that were asserting the improper behavior.

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json:1]: If option 1 is currently selected, I click the select dropdown arrow, mouse down to option 3, click the select dropdown arrow again. I see the option value does not change (expected) but it looks like a change event is emitted (unexpected)

dropdownactivation

Comment (atmgrifter00) [reply change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json:1]: Fixed.

**Review by rajsite — APPROVED** [review]

File: change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json (hunk 1 above)
Comment (jattasNI) [inline change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json:1]: While testing I noticed a small bug. It appears to be pre-existing so no need to fix in this PR.

1. filter mode is 'none'
2. tab focus the select
3. without opening, type 'z' and Zurich is correctly selected with a change event fired
4. tab away
5. shift tab back
6. Enter to open, arrow keys to change the value, Enter to commit
7. Now type 'z' again. Nothing happens.

LMK if you want a bug for this.

**Review by jattasNI — APPROVED** [review]

File: change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json (hunk 1 above)
Comment (atmgrifter00) [reply change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json:1]: Yes, please file a bug.

Comment (jattasNI) [reply change/@ni-nimble-components-b897b4de-cdc6-4a5c-84f7-e4a2fc2c9c07.json:1]: https://github.com/ni/nimble/issues/2082

**Pull request merged**